lib/parity/environment.rb

```diff
@@ -41,14 +41,25 @@ def backup
 
     def deploy
       if production?
-        Kernel.system("git push production master")
+        Kernel.system("git push production #{branch_ref}")
       else
         Kernel.system(
-          "git push #{environment} HEAD:master --force",
+          "git push #{environment} HEAD:#{branch_ref} --force",
         )
       end
     end
 
+    def branch_ref
+      main_ref = `find .git/refs/heads/main`
```

Comment:
It might be better to do this using [the built-in Git command for finding a ref](https://git-scm.com/docs/git-show-ref), just in case the file structure changes later:

```ruby
# These will be true/false
main_ref_exists = system("git show-ref --verify --quiet refs/heads/master")
master_ref_exists = system("git show-ref --verify --quiet refs/heads/main")

if main_ref_exists && !master_ref_exists
  "main"
else
  "master"
end
```